### DISCUSSION TITLE
Add County to QSO Panel under State

### DISCUSSION THREAD

**Discussion by magicbug** [magicbug/Cloudlog#850] [Ideas / Feature Requests]
Now that Station Profile can auto-complete some of the counties of the world and lots of US ops i know hunt them, we should probably add a similar drop-down to the QSO panel under State so these can be logged against the QSO, I think some logbook services will also provide this.

**Comment by AndreasK79** [comment]
It's coming.
![image](https://user-images.githubusercontent.com/6977712/106386515-9c83cf80-63d5-11eb-875e-b9e03294ac9d.png)
Added it to the form, and lookup is working. 

Now I need to save it to DB.

**Comment by AndreasK79** [comment]
#863 was added.

**Reply by AndreasK79** [reply]
I guess the next thing to do, is to add it to awards...

**Reply by magicbug** [reply]
Yes interestingly Marty N9EAT asked me about that the other day

**Reply by AndreasK79** [reply]
Seems to be pretty straightforward. Sum worked and confirmed. Then when clicking, show a list of counties with the possibility to click on county to get popup with QSOs.

**Reply by AndreasK79** [reply]
Work in progress:
![image](https://user-images.githubusercontent.com/6977712/107131401-2bad4d80-68d6-11eb-8af8-ae8669c2e9fe.png)

**Reply by AndreasK79** [reply]
Getting closer:
![image](https://user-images.githubusercontent.com/6977712/107147711-cdbd4c00-694f-11eb-8990-7c665a470dae.png)

**Reply by AndreasK79** [reply]
@magicbug PR submitted. Handling of counties need some changes. Change database field and saving of state+county is needed: https://www.adif.org/311/ADIF_311.htm#Secondary_Administrative_Subdivision

**Comment by magicbug** [comment]
Final task will be popular the county field also using data coming out of the callbook system but not a huge priority.

**Comment by AndreasK79** [comment]
It seems I've used the wrong field in the database for saving counties. COL_USACA_COUNTIES is meant to be used when a station is on the border between two counties. COL_CNTY is supposed to be used for a single county, 

I was trying to figure out why my log wasn't imported with counties. They are stored in COL_CNTY on import.

@magicbug should I supply a PR where I change field in the QSO dialog?
